Tests/RandomAccessCollectionTests.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import XCTest
+
+class RandomAccessCollectionTests: XCTestCase {
+    func testEnumerateWithNext() {
+        let ary = [1,2,3]
+        for (current, next) in ary.enumerateWithNext() {
+            XCTAssert(current == 1)
+            if let next = next {
+                XCTAssert(next == 2)
+            }
+            break
+        }
+    }
+
+    func testEnumerateWithPrevious() {
+        let ary = [1,2,3]
+        for (previous, current) in ary.enumerateWithPrevious() {
+            if current == 1 { continue }
+            XCTAssert(current == 2)
+            if let previous = previous {
```

Comment:
It would actually be the coolest if it would only iterate over the cases where there is a previous item
So it would iterate twice over an array of three. But wouldn't see how this could be done since you have to return something each iteration right?